### ISSUE TITLE
Party behind port forwarder uses wrong ip handler

### ISSUE THREAD

**Issue by frankcorneliusmartin** [TNO-MPC/communication#1]
The server port is send as a cookie to the client. It uses the port that is used when creating the `HTTPServer` for this: 
https://github.com/TNO-MPC/communication/blob/24b60e5561c4b145ff67ba358657d3602377da17/tno/mpc/communication/httphandlers.py#L63

We run the player behind a port forwarder, meaning that the port the `HTTPServer` listens too is not the same port the other parties can reach this server.  We hacked a way of changing this behaviour by monkey patching the `HTTPClient`: 

```python
def new_http_client_init(self, pool, addr: str, port: int,
                         ssl_ctx: Optional[ssl.SSLContext]):
    self.pool = pool
    self.addr = addr
    self.port = port
    self.ssl_ctx = ssl_ctx
    if self.pool.http_server is None:
        raise AttributeError("No HTTP Server initialized (yet).")
    # self.session
    cookies = {"server_port": str(self.pool.external_port)}
    if self.pool.loop.is_running():
        self.pool.loop.create_task(self._create_client_session(cookies))
    else:
        self.pool.loop.run_until_complete(self._create_client_session(cookies))
    self.msg_send_counter = 0
    self.total_bytes_sent = 0
    self.msg_recv_counter = 0
    self.send_lock = threading.Lock()
    self.recv_lock = threading.Lock()
    self.buffer = {}


tno.mpc.communication.httphandlers.HTTPClient.__init__ = new_http_client_init
```

Would it be possible to allow to set an external port when creating the `HTTPServer`?

**Comment by ThomasTNO** [comment]
Resolved in https://github.com/TNO-MPC/communication/commit/b8c96fd8fe01a5d33f5a3e28d0696be31f087709

**Comment by frankcorneliusmartin** [comment]
Would be nice if we could also specify the `external_port` when using the method `.add_http_server(...)`

**Comment by ThomasTNO** [comment]
Good suggestion, continued in #2

**Comment by ThomasTNO** [comment]
> Would be nice if we could also specify the `external_port` when using the method `.add_http_server(...)`

Resolved in 07072f10af608466067d19a3817ee38fbdd8b814
